pkg/obfuscator/ip.go

```diff
@@ -32,29 +33,50 @@ var (
 )
 
 type ipGenerator struct {
-	template   string
-	obfuscated string
-	count      int
+	replacements map[string]string
+	template     string
+	obfuscated   string
+	count        int
+	lock         sync.Mutex
 }
 
-func (g *ipGenerator) generateConsistent() string {
+func (g *ipGenerator) consistent(ip string) string {
+	g.lock.Lock()
+	defer g.lock.Unlock()
+
+	if r, ok := g.replacements[ip]; ok {
+		return r
+	}
+
 	g.count++
 	if g.count > maximumSupportedObfuscations {
 		klog.Exitf("maximum number of ip obfuscations exceeded: %d", maximumSupportedObfuscations)
 	}
-	return fmt.Sprintf(g.template, g.count)
+	r := fmt.Sprintf(g.template, g.count)
+	g.replacements[ip] = r
+	return r
 }
 
-func (g *ipGenerator) static() string {
+func (g *ipGenerator) static(ip string) string {
+	g.replacements[ip] = g.obfuscated
 	return g.obfuscated
 }
 
 type ipObfuscator struct {
-	ReplacementReporter
 	replacements    map[*regexp.Regexp]*ipGenerator
 	replacementType schema.ObfuscateReplacementType
 }
 
+func (o *ipObfuscator) ReportingResult() map[string]string {
+	result := make(map[string]string)
+	for _, replacers := range o.replacements {
```

Comment:
please also put that under the lock, you need the memory barrier otherwise you can read some funny cached mapping

Reply:
I didn't see an issue with this because this is called after the processing is actually done. But I see your point.

Reply:
Updated